mat64/io.go

```diff
@@ -5,14 +5,180 @@
 package mat64
 
 import (
+	"bytes"
 	"encoding/binary"
+	"errors"
+
+	"github.com/gonum/matrix"
 )
 
-var (
-	littleEndian  = binary.LittleEndian
-	bigEndian     = binary.BigEndian
-	defaultEndian = littleEndian
+// maxLen is the biggest slice/array len one can create on a 32/64b platform.
+const maxLen = int64(int(^uint(0) >> 1))
 
+var (
 	sizeInt64   = binary.Size(int64(0))
```

Comment:
I'm not sure if @kortschak was in favor of changing this. I think it is ok to leave as is.